scripts/fauxalysis_from_docking.py

```diff
@@ -1,37 +1,229 @@
+        return False
+
+    ## Then check for other intermediate files
+    du = oechem.OEDesignUnit()
+    if not oechem.OEReadDesignUnit(f"{d}/predocked.oedu", du):
+        return False
+
+    if load_openeye_pdb(f"{d}/predocked.pdb").NumAtoms() == 0:
+        return False
+
+    if load_openeye_sdf(f"{d}/docked.sdf").NumAtoms() == 0:
+        return False
+
+    return True
+
+
+# ToDo: Move this as well as other scripts to a more logical api
+def write_fragalysis_output(
+    in_dir, out_dir, best_structure_dict, frag_dir=None, cmpd_to_frag_dict=None
+):
+    """
+    Convert original-style output structure to a fragalysis-style output
+    structure.
+
+    Parameters
+    ----------
+    in_dir : str
+        Top-level directory of original-style output
+    out_dir : str
+        Top-level directory of new output
+    frag_dir : str
+        Path to fragalysis results directory.
+    best_structure_dict : dict
+        Of the form {('Compound_ID', dimer): 'Structure_Source'} where "dimer" is a boolean.
+    """
+    os.makedirs(f"{out_dir}", exist_ok=True)
+
+    ## Create list of sdf files we will make so that we can concatenate them at the end
+    cmpd_sdf_list = []
+
+    ## Loop through dict and parse input files into output files
+    for (compound_id, dimer), best_str in best_structure_dict.items():
+        ## Make sure input exists
+        dimer_s = "dimer" if dimer else "monomer"
+        compound_in_dir = f"{in_dir}/{compound_id}/{dimer_s}/{best_str}"
+        compound_out_dir = f"{out_dir}/{compound_id}"
+
+        ## If inputs don't exist, else if the output directory already exists, don't waste time
+        if not check_output(compound_in_dir):
+            print(
+                (
+                    f"No results found for {compound_id}/{dimer_s}/{best_str}, "
+                    "skipping"
+                ),
+                flush=True,
+            )
+            continue
+        else:
+            print(
+                (
+                    f"Generating fauxalysis...\n"
+                    f"\tfrom: {compound_in_dir}\n"
+                    f"\tto: {compound_out_dir}"
+                ),
+                flush=True,
+            )
+
+        ## Create output directory if necessary
+        os.makedirs(compound_out_dir, exist_ok=True)
+
+        ## Load necessary files
+        du = oechem.OEDesignUnit()
+        oechem.OEReadDesignUnit(f"{compound_in_dir}/predocked.oedu", du)
+        prot = oechem.OEGraphMol()
+        du.GetProtein(prot)
+        lig = load_openeye_sdf(f"{compound_in_dir}/docked.sdf")
+
+        ## Set ligand title, useful for the combined sdf file
+        lig.SetTitle(f"{compound_id}_{best_str}")
+
+        ## Give ligand atoms their own chain "L" and set the resname to "LIG"
+        for atom in lig.GetAtoms():
+            residue = oechem.OEAtomGetResidue(atom)
+            residue.SetChainID("L")
+            residue.SetName("LIG")
+            oechem.OEAtomSetResidue(atom, residue)
```

Comment:
I think if we add `residue.SetHetAtom(True)` before this line it should take care of #66